tests/Workflow/MoveJatsAssetsTest.php

```diff
@@ -0,0 +1,148 @@
+        $this->filesystem->addPlugin(new HttpUrlPlugin('http://assets/path/'));
+    }
+
+    /**
+     * @after
+     */
+    public function unregisterStreamWrapper() : void
+    {
+        FlysystemStreamWrapper::unregister('guzzle-cache');
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_it_is_not_jats() : void
+    {
+        $mover = new MoveJatsAssets('~^http://example.com/assets/~', $this->filesystem, $this->client);
+
+        $document = FluentDOM::load('<item xmlns="http://libero.pub"/>');
+
+        $marking = new Marking();
+        $task = new PutTask('service', ItemId::fromString('id'), ItemVersionNumber::fromInt(1), clone $document);
+        $transition = new Transition('transition', 'place1', 'place2');
+
+        $event = new Event($task, $marking, $transition);
+
+        $mover->onManipulate($event);
+
+        $this->assertXmlStringEqualsXmlString($document->saveXML(), $task->getDocument()->saveXML());
+        $this->assertEmpty($this->filesystem->listContents());
+    }
+
+    /**
+     * @test
+     */
+    public function it_handles_assets() : void
+    {
+        $mover = new MoveJatsAssets('~^http://www.example.com/assets/~', $this->filesystem, $this->client);
+
+        $document = FluentDOM::load(
+            <<<XML
+<item xmlns="http://libero.pub" xml:base="http://www.example.com/">
+    <article xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+        <body>
+            <graphic xlink:href="assets/figure1.jpg"/>
+        </body>
+    </article>
+</item>
+XML
+        );
+
+        $marking = new Marking();
+        $task = new PutTask('service', ItemId::fromString('id'), ItemVersionNumber::fromInt(1), $document);
+        $transition = new Transition('transition', 'place1', 'place2');
+
+        $event = new Event($task, $marking, $transition);
+
+        $this->mock->save(
+            new Request('GET', 'http://www.example.com/assets/figure1.jpg'),
+            new Response(200, ['Content-Type' => 'image/jpeg;foo=bar'], 'figure1')
+        );
+
+        $mover->onManipulate($event);
+
+        $path = 'id/v1/879f77a11b0649cb8af511fa5d6e4a7e.jpeg';
+
+        $expected = FluentDOM::load(
+            <<<XML
+<item xmlns="http://libero.pub" xml:base="http://www.example.com/">
+    <article xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+        <body>
+            <graphic xlink:href="http://assets/path/{$path}" mimetype="image" sub-mimetype="jpeg"/>
+        </body>
+    </article>
+</item>
+XML
+        );
+
+        $this->assertXmlStringEqualsXmlString($expected, $task->getDocument());
+        $this->assertTrue($this->filesystem->has($path));
+        $this->assertSame(AdapterInterface::VISIBILITY_PUBLIC, $this->filesystem->getVisibility($path));
+        $this->assertSame('figure1', $this->filesystem->read($path));
+        $this->assertSame('image/jpeg', $this->filesystem->getMimetype($path));
```

Comment:
This is actually guessing by the file extension rather than what's set, https://github.com/thephpleague/flysystem-memory/pull/6 will fix.